cmd/main.go

```diff
@@ -108,41 +120,42 @@ func execute(cmd *cobra.Command, args []string) {
 		os.Exit(1)
 	}
 
-	items := make([]plugins.Item, 0)
 	for _, plugin := range allPlugins {
 		if !plugin.IsEnabled() {
 			continue
 		}
 
-		pluginItems, err := plugin.GetItems()
-		if err != nil {
-			log.Fatal().Msg(err.Error())
-		}
-		items = append(items, *pluginItems...)
+		wg.Add(1)
+		go plugin.GetItems(itemsChannel, errorsChannel, &wg)
 	}
 
-	report := reporting.Report{}
-	report.Results = make(map[string][]reporting.Secret)
-
-	// -------------------------------------
-	// Detect Secrets
-
-	secrets := secrets.Init(tags)
-
-	for _, item := range items {
-		secrets := secrets.Detect(item.Content)
-		if len(secrets) > 0 {
-			report.TotalSecretsFound = report.TotalSecretsFound + len(secrets)
-			report.Results[item.ID] = append(report.Results[item.ID], secrets...)
+	go func() {
+		for {
+			select {
+			case item := <-itemsChannel:
+				report.TotalItemsScanned++
+				wg.Add(1)
+				go secrets.Detect(secretsChannel, item, &wg)
+			case secret := <-secretsChannel:
+				report.TotalSecretsFound++
+				report.Results[secret.ID] = append(report.Results[secret.ID], secret)
+			case err, ok := <-errorsChannel:
+				if !ok {
+					return
+				}
+				log.Fatal().Msg(err.Error())
+			}
 		}
-	}
-	report.TotalItemsScanned = len(items)
+	}()
+	wg.Wait()
+
+	// Wait for last secret to be added to report
+	time.Sleep(time.Millisecond * timeSleepInterval)
```

Comment:
Hi @joaopedrocsilva , very nice work, I added this to my PR #53.

I have one question... why we are sleeping here? Why the `waitGroup` is not helping us?

Reply:
Hey @baruchiro.

Couln't understand why but without the sleep we can't get all elements. 

Maybe it would make sense to spend some time investigating if it's realated to issues on goroutines management or it's related to some kind of limitations.